integration/clean/clean_aws_resource.go

```diff
@@ -0,0 +1,145 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build clean
+// +build clean
+
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Type "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	ssmType "github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	smithyTime "github.com/aws/smithy-go/time"
+)
+
+const (
+	daysToKeep   = 30
+	keepDuration = -1 * time.Hour * 24 * time.Duration(daysToKeep)
+)
+
+var expirationDate = time.Now().UTC().Add(keepDuration)
+
+func main() {
+	log.Println("Begin to clean EC2 AMI")
+	cleanAMI()
+
+	log.Println("Begin to clean SSM Parameter Store")
+	cleanSSMParameterStore()
+
+	log.Println("Finished cleaning resources.")
+}
+
+func cleanSSMParameterStore() {
+	ctx := context.Background()
+	defaultConfig, err := config.LoadDefaultConfig(ctx)
+
+	if err != nil {
+		log.Printf("Load default config failed because of %v", err)
+		return
+	}
+
+	ssmClient := ssm.NewFromConfig(defaultConfig)
+
+	//Allow to load all th since the default respond is paginated auto scaling groups.
+	//Look into the documentations and read the starting-token for more details
+	//Documentation: https://docs.aws.amazon.com/cli/latest/reference/autoscaling/describe-auto-scaling-groups.html#options
+	var nextToken *string
+
+	var parameterStoreNameFilter = ssmType.ParameterStringFilter{
+		Key:    aws.String("Name"),
+		Option: aws.String("BeginsWith"),
+		Values: []string{"AmazonCloudWatch"},
+	}
+
+	for {
+		describeParametersInput := ssm.DescribeParametersInput{
+			ParameterFilters: []ssmType.ParameterStringFilter{parameterStoreNameFilter},
+			NextToken:        nextToken,
+		}
+		describeParametersOutput, err := ssmClient.DescribeParameters(ctx, &describeParametersInput)
+
+		if err != nil {
+			log.Printf("Describe Parameter Stores failed because of %v", err)
+			return
+		}
+
+		for _, parameter := range describeParametersOutput.Parameters {
+
+			if !expirationDate.After(*parameter.LastModifiedDate) {
+				continue
+			}
+
+			log.Printf("Trying to delete Parameter Store with name %s and creation date %v", *parameter.Name, *parameter.LastModifiedDate)
+
+			deleteParameterInput := ssm.DeleteParameterInput{Name: parameter.Name}
+
+			if _, err := ssmClient.DeleteParameter(ctx, &deleteParameterInput); err != nil {
+				log.Printf("Failed to delete Parameter Store with name %s because of %v", *parameter.Name, err)
+				return
```

Comment:
My two issues with this are that:
1. Along the lines of what I said above, until we have a way to consistently alert us that there is an issue with this GitHub action, we should be capturing errors and failing the job.
2. This behavior is inconsistent even with the changes made in the AMI cleaning function. In that one, you kept the errors array, but then at the end check the length of the array to log something. 
Please update this to return an error array like it was originally

Reply:
Fixed the inconsistency between cleaning resources's behavior. Nice catch on this